discojs/discojs-core/src/dataset/data/image_data.ts

```diff
@@ -13,28 +13,32 @@ export class ImageData extends Data {
     // better error handling. An incorrectly formatted image in the dataset might still
     // cause an error during training, because of the lazy aspect of the dataset; we only
     // verify the first sample.
-    try {
-      const sample = (await dataset.take(1).toArray())[0]
-      // TODO: We suppose the presence of labels
-      // TODO: Typing (discojs-node/src/dataset/data_loader/image_loader.spec.ts)
-      if (!(typeof sample === 'object' && sample !== null)) {
-        throw new Error()
-      }
+    if (!task.trainingInformation.preprocessingFunctions?.includes(ImagePreprocessing.Resize) ||
+      !task.trainingInformation.RESIZED_IMAGE_H !== undefined ||
+      !task.trainingInformation.RESIZED_IMAGE_W !== undefined) {
```

Comment:
maybe a bit too many negations here?

```suggestion
  if (!(
    task.trainingInformation.preprocessingFunctions?.includes(ImagePreprocessing.Resize &&
    task.trainingInformation.RESIZED_IMAGE_H !== undefined &&
    task.trainingInformation.RESIZED_IMAGE_W !== undefined
  )) {
```

Reply:
yes, definitely!